### ISSUE TITLE
`reportAny` false negative on bare `ClassVar`

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#1515]
### Description

```py
from typing import ClassVar

class A:
    a: ClassVar # inferred as `ClassVar[Any]`
```

> If the bare `ClassVar` qualifier is used without any assigned value, the type should be inferred as [`Any`](https://typing.python.org/en/latest/spec/special-types.html#any). Type checkers may error if no assigned value is present.

- https://typing.python.org/en/latest/spec/class-compat.html#classvar

**Comment by DetachHead** [comment]
it should be implicit (`reportAny`) because it's inferred, not `reportExplicitAny`
